neet/boolean/sensitivity.py

```diff
@@ -484,3 +488,160 @@ def average_sensitivity(self, states=None, weights=None, calc_trans=True):
+                    # be 0, since a 0-hamming neighbor is just the
+                    # original state/vector
+                    print("This shouldn't print. 0-hamming neighbors should not diverge.")
+                    print("c: ", c)
+                    print("1. neighbor:  ", neighbors_copy[copy_counter])
+                    print("2. state:     ", state_copy,"\n")
+                    print("1. newState:  ", newState)
+                    print("2. nextState: ", nextState,"\n\n")
+                s += distance(newState, nextState)
+            copy_counter += 1
+            neighbor = c_hamming_neighbors(self,state,c)
+        return s / copy_counter
+
+
+    def Average_c_sensitivity(self, states=None, calc_trans=True, c=1):
+
+
+        """
+
+        :param states: a set of network states
+        :type states: list, numpy.ndarray
+        :param transitions: precomputed state transitions (*optional*)
+        :type transitions: list, numpy.ndarray, None
+        :param calc_trans: pre-compute all state transitions; ignored if
+                           ``states`` or ``weights`` is ``None``
+        :type calc_trans: bool
+        :return: the sensitivity averaged over all possible 
+                            states of the network
+
+
+        """
+
+
+        s = 0
+
+        if states is not None:
+            # optionally pre-calculate transitions
+            if calc_trans:
+                decoder = self.decode
+                trans = list(map(decoder, self.transitions))
+            else:
+                trans = None
+
+            # Iterate through all provided states and sum 
+            # the distances between them and their c-hamming-neighbors
+            # after a single time-step (one synchronous transition)
+            for state in states:
+                s += self.C_sensitivity_at_x(state, trans, c)
+
+            # get the average of s by diving the sum by the
+            # number of states considered
+            # ~Should this one be 2^len(states) or something else?~
+            s = s / np.power(2, len(states))
+            return s
+
+        else:
+            # Optionally pre-calculate transitions
+            if calc_trans:
+                decoder = self.decode
+                trans = list(map(decoder, self.transitions))
+            else:
+                trans = None
+            # Generate all possible states 
+            # and sum the distances between them and each of their
+            # c-hamming-neighbors in the next time step (after one 
+            # synchronous transition)
+            for n in range(self.size):
+                state_gen = itt.combinations(range(self.size),n)
+                for state in state_gen:
+                    state_array = [0 for x in range(self.size)]
+                    for index in state:
+                        state_array[index] = 1
+                    s += self.C_sensitivity_at_x(state_array, trans, c)
```

Comment:
All [Networks](https://neet.readthedocs.io/en/stable/api/network.html#network) are [StateSpace](https://neet.readthedocs.io/en/stable/api/statespace.html)s are [iterable](https://neet.readthedocs.io/en/stable/api/statespace.html#neet.StateSpace.__iter__), so you can just do
```python
for state in self:
    s += self.C_sensitivity_at_x(state, trans, c)
```